scripts/g.extension/g.extension.py

```diff
@@ -1555,7 +1586,7 @@ def download_source_code(
             response = urlopen(url)
         except URLError:
             # Try download add-on from 'master' branch if default "main" fails
-            if branch == "main":
+            if not branch:
```

Comment:
If the branch is coming from `get_default_branch`, checking main makes more sense to me here, but I can't figure out how branch is passed in the code. `resolve_known_host_service` seem to not use the parameter... Perhaps a topic for another PR.

Reply:
Yes. I think, in general, refactoring g.extension could be a very good task for the next community sprint (including improved git support and other issues and feaure requests that are open...)

Reply:
I already started to put some useful functions (originally taken from g.extension) to the library, so we are on a good track. The GitHub/GitLab API stuff could eventually go there too (later).